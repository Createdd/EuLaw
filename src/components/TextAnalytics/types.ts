interface TextAnalyticsPropsType {
    text: string;
    results: (res: TextAnalyticsStateType) => void;
}

interface TextAnalyticsStateType {
    topics: string[];
    places: string[];
}

export {
    TextAnalyticsPropsType,
    TextAnalyticsStateType,
};
