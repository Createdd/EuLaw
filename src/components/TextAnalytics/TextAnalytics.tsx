import * as React from 'react';
import { intersection as _intersection, isEqual as _isEqual } from 'lodash';
var nlp = require('compromise');

import { TextAnalyticsPropsType, TextAnalyticsStateType } from './types';
import TextAnalyticsStyled from './styles';

import ExampleCase from '../../data/exampleData/exampleCase';
import MemberStates from '../../data/exampleData/memberStates';

class TextAnalytics extends React.Component<
  TextAnalyticsPropsType,
  TextAnalyticsStateType
> {
  constructor(props: TextAnalyticsPropsType) {
    super(props);
    this.state = {
      topics: [],
      places: []
    };
  }

  componentDidMount() {
    this.calculateResults();
  }

  resultsHandler = (res: TextAnalyticsStateType) => {
    this.props.results(res);
  }

  calculateResults = () => {
    // const text = nlp(this.props.text); // TODO: Use input instead of example
    const text = nlp(ExampleCase.text);
    const topics = text
      .topics()
      .slice(0, 4)
      .out('frequency')
      .map((top: { normal: string }) => top.normal.toLowerCase());
    const places = text
      .nouns()
      .places()
      .slice(0, 4)
      .out('frequency')
      .map((pl: { normal: string }) => pl.normal.toLowerCase());

    if (!_isEqual(this.state, { topics, places })) {
      this.setState({
        topics,
        places
      });
    }

    const intersect = _intersection(
      MemberStates.map((str: string) => str.toLowerCase()),
      places
    );

    const newRes = {
      topics: [],
      places: intersect
    };

    this.resultsHandler(newRes);
  }

  renderResults = () => {
    this.calculateResults();
    return (
      <ul>
        <li>
          Topics:
          {this.state.topics.map((top: string) => `${top.toUpperCase()}, `)}
        </li>
        <li>
          States:
          {this.state.places.map((pl: string) => `${pl.toUpperCase()}, `)}
        </li>
      </ul>
    );
  }

  render() {
    return (
      <TextAnalyticsStyled>
        <p>TextAnalytics</p>
        {this.renderResults()}
      </TextAnalyticsStyled>
    );
  }
}

export { TextAnalytics as default };
