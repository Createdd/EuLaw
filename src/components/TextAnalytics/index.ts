export { default } from './TextAnalytics';
export {
    TextAnalyticsPropsType,
    TextAnalyticsStateType,
} from './types';
