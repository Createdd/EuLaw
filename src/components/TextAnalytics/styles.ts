import styled from 'styled-components';

const TextAnalyticsStyled = styled.div`
  ul {
    margin-left: 5rem;
    text-align: left;
    list-style-position: inside;
  }
`;

export { TextAnalyticsStyled as default };
