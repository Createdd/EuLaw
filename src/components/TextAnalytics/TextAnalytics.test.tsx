import * as React from 'react';
import * as renderer from 'react-test-renderer';

import TextAnalytics from './TextAnalytics';

const testText = 'test';

describe( 'Test TextAnalytics', () => {
    it( 'renders snapshot', () => {
        const tree = renderer.create(
            <TextAnalytics text={testText} />
        ).toJSON();

        expect( tree ).toMatchSnapshot();
    } );
} );
