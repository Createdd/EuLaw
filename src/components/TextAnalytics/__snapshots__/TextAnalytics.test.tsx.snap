// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test TextAnalytics renders snapshot 1`] = `
<div
  className="sc-bdVaJa CARIw"
>
  <p>
    TextAnalytics
  </p>
  <ul>
    <li>
      euTopic:
    </li>
    <li>
      States:
    </li>
  </ul>
</div>
`;
